crates/core/src/db/datastore/locking_tx_datastore/mod.rs

```diff
@@ -2503,165 +2632,99 @@ mod tests {
+        ]));
         #[rustfmt::skip]
-        assert_eq!(
-            column_rows,
-            vec![
-                // table_id, col_id, col_name, col_type, is_autoinc
-                column_row(0, 0, "table_id", AlgebraicType::U32, true),
-                column_row(0, 1, "table_name", AlgebraicType::String, false),
-                column_row(0, 2, "table_type", AlgebraicType::String, false),
-                column_row(0, 3, "table_access", AlgebraicType::String, false),
-
-                column_row(1, 0, "table_id", AlgebraicType::U32, false),
-                column_row(1, 1, "col_id", AlgebraicType::U32, false),
-                column_row(1, 2, "col_type", AlgebraicType::bytes(), false),
-                column_row(1, 3, "col_name", AlgebraicType::String, false),
-                column_row(1, 4, "is_autoinc", AlgebraicType::Bool, false),
-
-                column_row(2, 0, "sequence_id", AlgebraicType::U32, true),
-                column_row(2, 1, "sequence_name", AlgebraicType::String, false),
-                column_row(2, 2, "table_id", AlgebraicType::U32, false),
-                column_row(2, 3, "col_id", AlgebraicType::U32, false),
-                column_row(2, 4, "increment", AlgebraicType::I128, false),
-                column_row(2, 5, "start", AlgebraicType::I128, false),
-                column_row(2, 6, "min_value", AlgebraicType::I128, false),
-                column_row(2, 7, "max_value", AlgebraicType::I128, false),
-                column_row(2, 8, "allocated", AlgebraicType::I128, false),
-
-                column_row(3, 0, "index_id", AlgebraicType::U32, true),
-                column_row(3, 1, "table_id", AlgebraicType::U32, false),
-                column_row(3, 2, "cols", AlgebraicType::array(AlgebraicType::U32), false),
-                column_row(3, 3, "index_name", AlgebraicType::String, false),
-                column_row(3, 4, "is_unique", AlgebraicType::Bool, false),
-
-                column_row(4, 0, "constraint_id", AlgebraicType::U32, true),
-                column_row(4, 1, "constraint_name", AlgebraicType::String, false),
-                column_row(4, 2, "kind", AlgebraicType::U32, false),
-                column_row(4, 3, "table_id", AlgebraicType::U32, false),
-                column_row(4, 4, "columns", AlgebraicType::array(AlgebraicType::U32), false),
-
-                column_row(5, 0, "program_hash", AlgebraicType::array(AlgebraicType::U8), false),
-                column_row(5, 1, "kind", AlgebraicType::U8, false),
-                column_row(5, 2, "epoch", AlgebraicType::U128, false),
-            ]
-        );
-        let index_rows = datastore
-            .iter_mut_tx(&ExecutionContext::default(), &tx, ST_INDEXES_ID)?
-            .map(|x| StIndexRow::try_from(x.view()).unwrap().to_owned())
-            .sorted_by_key(|x| x.index_id)
-            .collect::<Vec<_>>();
+        assert_eq!(query.scan_st_columns()?, map_array([
```

Comment:
It seems like there's two kinds of changes to all this test code.

One is fairly mechanical changes due to the renames etc that make up the rest of this PR.

The other is a substantial change to how the datastore exposes system tables to test code.

Is there any other kind of change that happens in this test code?